server/bin/pbench-unpack-tarballs.sh

```diff
@@ -37,6 +37,21 @@
 # symlink from $ARCHIVE/TODO to $ARCHIVE/TO-COPY-SOS.
 
 
+BUCKET="${1}"
+PIPELINE="${2}"
+
+if [[ "${PIPELINE}" == "re-unpack" ]]; then
+    export PROG="pbench-re-unpack-tarballs"
+fi
+if [[ -z "${BUCKET}" || "${BUCKET}" == "none" ]]; then
+    export PROG="${PROG}"
+else
+    # We rename the PROG to include the bucket since we don't want to conflict
+    # with other unpack tar balls running using different buckets at the same
+    # time.
+    export PROG="${PROG}-${BUCKET}"
+fi
```

Comment:
What exactly does line 47 accomplish?  (I would really appreciate a code comment there....)

And, are we sure that we don't want to combine lines 45 & 46 into an `elif`?

Reply:
> And, are we sure that we don't want to combine lines 45 & 46 into an `elif`?

Yes, they are orthogonal considerations.

The update to line 40 makes this easier to shrink the logic, though.